src/components/ChatSection/ChatWindow/ChatWindow.js

```diff
@@ -1,141 +1,124 @@
-                    this.arrRefs[index] = {
-                        ref: ref,
-                        message: item,
-                        dialogId: this.dialog.dialogId
-                    }
-                } }>
-                    <ChatItem
-                        messageUserId={ this.messageUserId }
-                        item={ item }
-                        interlocutor={ interlocutor }
-                        currentUser={ this.props.currentUser }
-                        modifyTime={ this.modifyTime }
-                    />
-                </div>
+        return chatMessages.map(messageItem => {
+            const isSameMessageOwner = prevMessageUserId.current === messageItem.userId;
+            prevMessageUserId.current = messageItem.userId;
+            return (
+                <InView as="div" onChange={ (inView, entry) => observeChat(inView, entry, messageItem, lastReadMessage, lastMessage) } key={ messageItem.id }>
+                    <div>
+                        <ChatItem
+                            isSameMessageOwner={ isSameMessageOwner }
+                            avatar={ currentUser.id === messageItem.userId ? currentUser.avatar : interlocutor.avatar }
+                            name={ currentUser.id === messageItem.userId ? currentUser.name : interlocutor.name }
+                            messageItem={ messageItem }
+                            currentUserId={ currentUser.id }
+                        />
+                    </div>
+                </InView>
+            )
+        });
+    };
 
-            this.messageUserId = item.userId
-            return chatItem
-        })
+    return (
+        <div className={ classes.chats_window }>
+            { preloader && <DialogPreloader/> }
+            <div className={ classes.chat_wrapper }>
+                <div ref={ chatStart }/>
+                { getUserChat() }
+                <div ref={ chatEnd }/>
+            </div>
 
-        this.messageUserId = null
-        return resultDialog
-    }
+            {
+                !lastMessageIsVisible && !!currentChat.current.messages.length &&
+                <div className={ classes.arrowDown_wrap }>
+                    { !!currentChat.current.unreadMessages &&
+                    <span className={ classes.sum_unread_messages }>{ currentChat.current.unreadMessages }</span> }
+                    <img src={ arrowDown } className={ classes.arrow_down } onClick={ smoothScrollDown } alt=""/>
+                </div>
+            }
+        </div>
+    )
+};
 
-    render() {
-        return (
-            <div className={ classes.chats_window }>
-                {
-                    !this.props.match.params.dialogId ?
-                        <span className={ classes.start_text }>Select a dialog and start communication</span> :
+const mapStateToProps = (state) => ({
+    dialogsState: state.dialogsDataReducer,
+    currentUser: state.dialogsDataReducer.currentUser,
```

Comment:
you can pass to the props whole dialogsDaaReducer. It contains currentUser in it